modulus/datapipes/climate/era5_hdf5.py

```diff
@@ -185,7 +197,11 @@ def parse_dataset_files(self) -> None:
             # truncate the dataset to avoid out-of-range sampling and ensure each
             # rank has same number of samples (to avoid deadlocks)
             data_samples_per_year = (
-                (f["fields"].shape[0] - self.num_steps * self.stride) // self.world_size
+                (
```

Comment:
Are we absolutely sure that `f["fields"].shape[0]` is always >= `(self.num_steps + self.num_history) * self.stride` so you don't get negative values?

Reply:
Yeah in the past this caused some annoying errors.

Reply:
Added an exception to check this